### ISSUE TITLE
support hammerdbws without a tty

### ISSUE THREAD

**Issue by grooverdan** [TPC-Council/HammerDB#572]
**Guidance**
Bug reports are for when HammerDB is not behaving as expected.
Bug reports should not be submitted for help in understanding database performance related questions.
General questions on database performance or HammerDB usability should be submitted under Discussions.

**Describe the bug**

Container services typically run without a tty. Running hammerdbws as a container seems very useful

But attempting to start it as so rejects because of the lack of a tty.

**To Reproduce**

Steps to reproduce the behaviour:
```
$ podman run --rm --publish 8080:8080 --entrypoint ./hammerdbws  docker.io/tpcorg/hammerdb:base   
HammerDB Web Service v4.7
Copyright (C) 2003-2023 Steve Shaw
Type "help" for a list of commands
Initialized new SQLite on-disk database /tmp/hammer.DB
Starting HammerDB Web Service on port 8080
Listening for HTTP requests on TCP port 8080
stty: impossible in this context
are you disconnected or in a batch, at, or cron script?stty: ioctl(user): bad file number

    while executing
"stty raw -echo"
    (procedure "rawInput" line 7)
    invoked from within
"rawInput"
    (procedure "TclReadLine::interact" line 39)
    invoked from within
"TclReadLine::interact"
    (procedure "TclReadLine::interactws" line 3)
    invoked from within
"TclReadLine::interactws"
    invoked from within
"if { $autostart::autostartap == "true" } {
        start_webservice -nowait
        source $autostart::autoloadscript
        } else {
        start_w..."
    (file "/home/hammerdb/HammerDB-4.7/./hammerdbws" line 83)
```

**Expected behavior**

```
$ podman run --rm --publish 8080:8080 --entrypoint ./hammerdbws -t  docker.io/tpcorg/hammerdb:base   
HammerDB Web Service v4.7
Copyright (C) 2003-2023 Steve Shaw
Type "help" for a list of commands
Initialized new SQLite on-disk database /tmp/hammer.DB
Starting HammerDB Web Service on port 8080
Listening for HTTP requests on TCP port 8080
hammerws>
```

This is forcing a tty there. But without a tty I'd just expect the webservice to do perform logging on the output.

**Screenshots**

**HammerDB Version (please complete the following information):**
 - Version: 4.7
 - Build: any container

**HammerDB Interface (please complete the following information):**
 - UI: none :-)

**Operating System (please complete the following information):**

Within podman container environment running ubuntu:focal.

**Database Server (please complete the following information):**
- didn't get that far

**Database Client (please complete the following information):**
- none
- 
**Additional context**

nfi.

**Comment by sm-shaw** [comment]
This 'should' be fixed in v4.8 with the changed webservice to provide the jobs interface with this pull request https://github.com/TPC-Council/HammerDB/pull/558
From v4.8 the webservice can be started manually or from the CLI itself or the GUI.
The example is from Windows but Linux is the same:
Using no argument or wait, the web service will only listen for HTTP requests and not return a prompt.
```
C:\Program Files\HammerDB-4.8>hammerdbws wait
HammerDB Web Service v4.8
Copyright (C) 2003-2023 Steve Shaw
Type "help" for a list of commands
Starting HammerDB Web Service on port 8080
```
With nowait it will return a prompt and the HTML can be fetched from the prompt. (The HTML is stripped for the help command that shows the commands accepted. 
```
C:\Program Files\HammerDB-4.8>hammerdbws nowait
HammerDB Web Service v4.8
Copyright (C) 2003-2023 Steve Shaw
Type "help" for a list of commands
Starting HammerDB Web Service on port 8080
hammerws>jobs
<html>
<head>
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta http-equiv="content-type" content="text/html; charset=UTF-8">
<link href="http://localhost:8080/style.css" rel="stylesheet">
<title>HammerDB Results</title>
</head>
<body>
<p><img src='http://localhost:8080/logo.png' width='347' height='60'></p><h3 class="title">Job Index:</h3><div class='hammerdb' data-title='Jobs'><div><ol style='column-width: 20ex;'>
...
hammerws>help
HammerDB Web Service
Help:
GET jobs: Show the job ids, configuration, output, status, results and timings of jobs created by buildschema and vurun. Job output is equivalent to the output viewed in the graphical interface or command line.
get http://localhost:8080/jobs
get http://localhost:8080/jobs?jobid=TEXT
get http://localhost:8080/jobs?jobid=TEXT&bm
get http://localhost:8080/jobs?jobid=TEXT&db
get http://localhost:8080/jobs?jobid=TEXT&delete
get http://localhost:8080/jobs?jobid=TEXT&dict
get http://localhost:8080/jobs?jobid=TEXT&index
get http://localhost:8080/jobs?jobid=TEXT&result
get http://localhost:8080/jobs?jobid=TEXT&resultdata
get http://localhost:8080/jobs?jobid=TEXT&status
get http://localhost:8080/jobs?jobid=TEXT&tcount
get http://localhost:8080/jobs?jobid=TEXT&tcountdata
get http://localhost:8080/jobs?jobid=TEXT&timestamp
get http://localhost:8080/jobs?jobid=TEXT&timing
get http://localhost:8080/jobs?jobid=TEXT&timingdata
get http://localhost:8080/jobs?jobid=TEXT&timing&vuid=INTEGER
get http://localhost:8080/jobs?jobid=TEXT&vu=INTEGER
hammerws>
```
There is also the HammerDB GUI argument (as used on the GUI) that just listens and returns no output at all to the console.
`C:\Program Files\HammerDB-4.8>hammerdbws gui` this can be useful when called by other commands. 

In the CLI there is the wsstart, wsstop, wsstatus and wsport commands, with the latter either reporting or setting the port to listen on.
```
HammerDB CLI v4.8
Copyright (C) 2003-2023 Steve Shaw
Type "help" for a list of commands
Initialized Jobs on-disk database C:/Users/Steve/AppData/Local/Temp/hammer.DB using existing tables (245,760 KB)
hammerdb>wsstart
HammerDB Web Service v4.8
Copyright (C) 2003-2023 Steve Shaw
Type "help" for a list of commands
Starting HammerDB Web Service on port 8080
hammerdb>wsstop
Stopping HammerDB Web Service on port 8080
hammerdb>wsstatus
Web Service not running: connect failed connection refused
hammerdb>wsport
Web Service Port set to 8080
```

So from v4.8 either nowait or gui prompts should allow the web service to run without the tty. 

Just an additional point on windows is that sometimes the security features prompt for a port to be open that is not received if running the GUI so you have to start manually or from the CLI to then get and allow the port prompt.

**Comment by sm-shaw** [comment]
This has been included in the v4.8 docker release
